### PR TITLE
C2PA-307: Fix SFNT checksum calculation

### PR DESCRIPTION
**Pull request by chriskushmonotype**
## Changes in this pull request

Fixes the calculation of the SFNT checksum.

## Checklist
- [x] This PR represents a single feature, fix, or change.
- [ ] All applicable changes have been documented.
- [ ] Any `TO DO` items (or similar) have been entered as GitHub issues and the link to that issue has been included in a comment.

### CODE REVIEW

File: sdk/tests/fixtures/font_c2pa.otf
Comment (chriskushmonotype) [inline sdk/tests/fixtures/font_c2pa.otf:1]: @vancejc-mt Remind me that I need to re-generate this using a less-informal test manifest 😇

File: sdk/src/asset_handlers/font_io.rs [hunk 1]
```diff
@@ -212,6 +213,27 @@ pub(crate) fn checksum(bytes: &[u8]) -> Wrapping<u32> {
...
+/// Wrapping<u32> with the data checksum. (Note that trailing pad bytes do not
+/// affect this checksum - it's not a real CRC.)
+#[allow(dead_code)]
+pub(crate) fn checksum_biased(bytes: &[u8], bias: u32) -> Wrapping<u32> {
+    match bias & 3 {
+        0 => checksum(bytes),
+        1 => Wrapping(BigEndian::read_u24(bytes)) + checksum(&(bytes[3..bytes.len()])),
+        2 => Wrapping(BigEndian::read_u16(bytes) as u32) + checksum(&(bytes[2..bytes.len()])),
+        3 => Wrapping(bytes[0] as u32) + checksum(&(bytes[1..bytes.len()])),
+        4_u32..=u32::MAX => todo!(),
```
Comment (vancejc-mt) [inline sdk/src/asset_handlers/font_io.rs:233]: How.. would you get a 4 on a module 3, overflow?

File: sdk/src/asset_handlers/font_io.rs [hunk 2]
```diff
@@ -310,15 +332,32 @@ impl TableC2PARaw {
...
-            manifestStoreLength: reader.read_u32::<BigEndian>()?,
+            majorVersion: maj,
+            minorVersion: min,
+            activeManifestUriOffset: uo,
+            activeManifestUriLength: us,
+            reserved: rs,
+            manifestStoreOffset: mo,
+            manifestStoreLength: ms,
         })
+        //Ok(Self {
```
Comment (vancejc-mt) [inline sdk/src/asset_handlers/font_io.rs]: Did you wanna leave this comment in?

**Review by vancejc-mt — APPROVED** [review]
Works like a charm (tested with Algerian with the test manifest from `c2patool` using FontValidator); just some nit questions

File: sdk/src/asset_handlers/font_io.rs (hunk 1 above)
Comment (chriskushmonotype) [reply sdk/src/asset_handlers/font_io.rs:233]: Right, I'm finally running into corners of Rust like this (and like the things `.chunks_exact` kicks back being runtime-sized).

Comment (brogdonm) [reply sdk/src/asset_handlers/font_io.rs:233]: This kind of thing should be an open issue/request on the main language implementation (if not already).

**Pull request merged**